### DISCUSSION TITLE
mspi Flash NOR device giving size to the MSPI controller

### DISCUSSION THREAD

**Discussion by FRASTM** [zephyrproject-rtos/zephyr#78997] [Q&A]
Our stm32 octospi controller requires the size of the device it is talking to

In the configuration where the  octo flash NOR mx25lm51245  has a <size> property, this value must be passed ot the Parent instance for its initialization at device init. 

```
&mspi1 {
 compatible = "st,stm32-mspi-controller";
 ...
   mx25lm51245: ospi-nor-flash@90000000 {
  		compatible = "mspi-nor-mx25";
		size = <DT_SIZE_M(64)>; /* in Bytes */
   }
};
```

What would be the structure to pass the    flash_mspi_nor_mx_config.mem_size  to the MSPI controller ?
What would hold that value, there is no field in the mspi_dev_cfg  or mspi_cfg or mspi_dev_id

Note that The mspi_xip_cfg  is probably the best place to hold the size (and base address)  from the drivers/flash  to the drivers/mspi, but this is coming too late in the system init or not coming at all.

**Comment by erwango** [comment]
@swift-tk Any idea ?
(^^@anangl)
